cmd/bootstrap/cmd/key.go

```diff
@@ -9,47 +9,108 @@ import (
+	"github.com/onflow/flow-go/crypto"
 	model "github.com/onflow/flow-go/model/bootstrap"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/network/p2p"
 )
 
 var (
-	flagRole        string
-	flagAddress     string
+	flagRole    string
+	flagAddress string
+
+	// bools
+	flagNetworkKey bool
+	flagStakingKey bool
+	flagMachineKey bool
+
+	// seed flags
 	flagNetworkSeed []byte
 	flagStakingSeed []byte
+	flagMachineSeed []byte
 )
 
+// keyCmd represents the key command
+var keyCmd = &cobra.Command{
+	Use:   "key",
+	Short: "Generate networking and staking keys for a partner node and write them to files",
+	Run:   keyCmdRun,
+}
+
+func init() {
+	rootCmd.AddCommand(keyCmd)
+
+	// required flags
+	keyCmd.Flags().StringVar(&flagRole, "role", "", "node role (can be \"collection\", \"consensus\", \"execution\", \"verification\" or \"access\")")
+	_ = keyCmd.MarkFlagRequired("role")
+	keyCmd.Flags().StringVar(&flagAddress, "address", "", "network address")
+	_ = keyCmd.MarkFlagRequired("address")
+
+	keyCmd.Flags().BoolVar(&flagNetworkKey, "networking", false, "generate networking key only")
+	keyCmd.Flags().BoolVar(&flagNetworkKey, "staking", false, "generate staking key only")
+	keyCmd.Flags().BoolVar(&flagNetworkKey, "machine", false, "generate machine key only")
+
+	keyCmd.Flags().BytesHexVar(&flagNetworkSeed, "networking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded networking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagStakingSeed, "staking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded staking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagMachineSeed, "machine-seed", generateRandomSeed(), fmt.Sprintf("hex encoded machine account seed (min %v bytes)", minSeedBytes))
+}
+
 // keyCmdRun generate the node staking key, networking key and node information
 func keyCmdRun(_ *cobra.Command, _ []string) {
 	// validate inputs
 	role := validateRole(flagRole)
 	validateAddressFormat(flagAddress)
-	networkSeed := validateSeed(flagNetworkSeed)
-	stakingSeed := validateSeed(flagStakingSeed)
 
-	log.Debug().Msg("will generate networking key")
-	networkKeys, err := run.GenerateNetworkingKeys(1, [][]byte{networkSeed})
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate networking key")
+	var networkKey crypto.PrivateKey
+	var stakingKey crypto.PrivateKey
+	var machineKey crypto.PrivateKey
+
+	if !flagNetworkKey && !flagStakingKey && !flagMachineKey {
+		flagNetworkKey = true
+		flagStakingKey = true
+		flagMachineKey = true
 	}
-	log.Info().Msg("generated networking key")
 
-	log.Debug().Msg("will generate staking key")
-	stakingKeys, err := run.GenerateStakingKeys(1, [][]byte{stakingSeed})
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate staking key")
+	if flagNetworkKey {
+		log.Debug().Msg("will generate networking key")
+		networkSeed := validateSeed(flagNetworkSeed)
+		networkKeys, err := run.GenerateNetworkingKeys(1, [][]byte{networkSeed})
```

Comment:
There are singular versions of these key generation functions we can use instead. (Likewise for the staking key generation.)
```suggestion
		networkKey, err := run.GenerateNetworkingKey(networkSeed)
```